webui/src/Buttons/EditButton.tsx

```diff
@@ -506,17 +568,15 @@ function TabsSection({ style, controlId, location, steps, runtimeProps, rotaryAc
 								// both selected and the current step
 								const isActiveAndCurrent = k === selectedIndex && runtimeProps.current_step_id === k
 
+								const name = steps[k]?.name;
+
 								if (moreThanOneStep) {
 									if (isActiveAndCurrent) linkClassname = 'selected-and-active'
 									else if (isCurrent) linkClassname = 'only-current'
 								}
 
 								return (
-									<CNavItem key={k} className="nav-steps-special">
-										<CNavLink data-tab={`step:${k}`} className={linkClassname}>
-											{i === 0 ? (keys.length > 1 ? 'Step ' + (i + 1) : 'Actions') : i + 1}
-										</CNavLink>
-									</CNavItem>
+									<TabStep selectedKey={selectedKey} k={k} i={i} name={name} inputState={{showInputEle:false}} linkClassname={linkClassname} socket={socket} controlId={controlId} ></TabStep>
```

Comment:
the `key` property should remain at this level, rather than being pushed inside. It is an 'id' field for react to handle arrays/lists of similar components https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key